src/probnum/kernels/_linear.py

```diff
@@ -0,0 +1,54 @@
+"""Linear covariance function."""
+
+from typing import Optional
+
+import numpy as np
+
+import probnum.utils as _utils
+from probnum.type import IntArgType, ScalarArgType
+
+from ._kernel import Kernel
+
+_InputType = np.ndarray
+
+
+class Linear(Kernel[_InputType]):
+    """Linear kernel.
+
+    Linear covariance function defined by :math:`k(x_0, x_1) = x_0^\\top x_1 + c`.
```

Comment:
just to make sure: you did not want a shift here, as in `(x1 - a)(x2-a) + b`? I mean, this is also something that can be done in a future PR, I just want to figure out the objective :)

Reply:
We may want this in the future when we have optimization of kernel hyperparameters, but for now this just corresponds to a fixed shift of the data, which can be part of the preprocessing pipeline.